sunpy/net/scraper_utils.py

```diff
@@ -77,33 +86,66 @@ def date_floor(date, timestep):
 
     return datetime(*time_tup)
 
-def check_timerange(pattern, url, timerange):
+def extractDateURL(self, url):
     """
-    Checks whether the time range represented in *url* intersects
-    with the given time range.
-
-    Parameters
-    ----------
-    url : `str`
-        URL of the file.
-    timerange : `~sunpy.time.TimeRange`
-        Time interval for which files were searched.
-
-    Returns
-    -------
-    `bool`
-        `True` if URL's valid time range overlaps the given timerange, else `False`.
+    Extracts the date from a particular url following the pattern.
     """
-    exdict = parse(pattern, url).named
-    if exdict['year'] < 100:
-        exdict['year'] = 2000 + exdict['year']
-    if 'month' not in exdict:
-                if 'month_name' in exdict:
-                    exdict['month'] = datetime.strptime(exdict['month_name'], '%B').month
-                elif 'month_name_abbr' in exdict:
-                    exdict['month'] = datetime.strptime(exdict['month_name_abbr'], '%b').month
-    tr = get_timerange_from_exdict(exdict)
-    return tr.intersects(timerange)
+    # remove the user and passwd from files if there:
+    url = url.replace("anonymous:data@sunpy.org@", "")
```

Comment:
Can a user supply different credentials and if so would this break?

Reply:
i cant think of an example that'd hit this but i don't think it should break, this was most likely originally added since the current implementation [overrides ftp filelist-ing with these credentials]( https://github.com/sunpy/sunpy/blob/scraper_rewrite/sunpy/net/scraper.py#L207) and this function internally gets called later on.

Generalising it would be a simple regex change though, let me know if it seems like a good change to have.

However I'm a bit skeptic about this check ever passing since after looking up FTP URL structure the second `@` shouldn't be there?